tests/Microsoft.Graph.Test/Requests/Functional/SharePointTests.cs

```diff
@@ -99,14 +102,67 @@ public class SharePointTests : GraphTestBase
         {
             try
             {
-                // 
-                Site site = graphClient.Shares[UrlToSharingToken("https://mod810997.sharepoint.com/sites/SMBverticals")].Site.Request().GetAsync().Result;
+                // Create the request to get the root site by using the root structural property. We don't generate 
+                // request builders for structural properties so we need to use HttpRequestMessage to make the request.
+                string requestUrlToGetSiteRootInfo = String.Format("{0}{1}", graphClient.Sites.Request().RequestUrl, "/root");
+                HttpRequestMessage hrm = new HttpRequestMessage(HttpMethod.Get, requestUrlToGetSiteRootInfo);
+
+                // Authenticate (add access token) to our HttpRequestMessage
+                await graphClient.AuthenticationProvider.AuthenticateRequestAsync(hrm);
+
+                HttpResponseMessage response = await graphClient.HttpProvider.SendAsync(hrm);
+
+                Site site;
+
+                // Get the Site.
+                if (response.IsSuccessStatusCode)
+                {
+                    // Deserialize Site object.
+                    var content = await response.Content.ReadAsStringAsync();
+                    site = graphClient.HttpProvider.Serializer.DeserializeObject<Site>(content);
+                }
+                else
+                    throw new ServiceException(
+                        new Error
+                        {
+                            Code = response.StatusCode.ToString(),
+                            Message = await response.Content.ReadAsStringAsync()
+                        });
+                               
+                string siteResource = "portals/Information-Technology";
+
+                // Get the portals/Information-Technology site.
+                Site portalInfoTechSite = await graphClient.Sites.GetSiteByPath(site.SiteCollection.Hostname, siteResource).Request().GetAsync();
+
+                StringAssert.Contains(portalInfoTechSite.WebUrl, siteResource);
+                StringAssert.Contains(portalInfoTechSite.Id, portalInfoTechSite.SiteCollection.Hostname); // Check if id format changes under us. 
+                Assert.AreEqual(site.SiteCollection.Hostname, portalInfoTechSite.SiteCollection.Hostname);
+            }
+            catch (Microsoft.Graph.ServiceException e)
+            {
+                Assert.Fail("Something happened, check out a trace. Error code: {0}", e.Error.Code);
+            }
+        }
+        
+        /// <summary>
+        /// Test to get information about a SharePoint site by its URL.
+        /// </summary>
+        [TestMethod]
+        public async Async.Task SharePointAccessSiteByUrl()
+        {
+            try
+            {
+                Site site = await graphClient.Shares[UrlToSharingToken("https://mod810997.sharepoint.com/sites/SMBverticals")].Site.Request().GetAsync();
```

Comment:
Can we pull this into a constants file? It looks like this is hard coded to one tenant.